src/6_branch.js

```diff
@@ -1965,3 +1965,36 @@ Branch.prototype['trackCommerceEvent'] = wrap(callback_params.CALLBACK_ERR, func
 	});
 	done();
 });
+/*
+ logEvent() documentation
+ */
+Branch.prototype['logEvent'] = wrap(callback_params.CALLBACK_ERR, function(done, name, eventAndCustomData, contentItems) {
+	name = name && utils.validateParameterType(name, 'string') ? name : null;
+	eventAndCustomData = eventAndCustomData && utils.validateParameterType(eventAndCustomData, 'object') ? eventAndCustomData : null;
+	contentItems = contentItems && utils.validateParameterType(contentItems, 'array') ? contentItems : null;
+
+	var endpoint = utils.determineV2EventType(name);
+	var extractedEventAndCustomData = utils.separateEventAndCustomData(eventAndCustomData);
+
+	var eventCallback = function(err, data) {
+		return done(err || null);
+	};
+
+	if (endpoint === 'STANDARD') {
+		this._api(resources.logStandardEvent, {
+			"name": name,
+			"custom_data": safejson.stringify(extractedEventAndCustomData && extractedEventAndCustomData['custom_data'] || {}),
```

Comment:
Spoke with the team about this and they would like an empty object sent up. Name is required so if it's null our validation rule will take over and throw an error.